swarm/api/api.go

```diff
@@ -204,52 +200,55 @@ func (m *MultiResolver) getResolveValidator(name string) ([]ResolveValidator, er
 	return rs, nil
 }
 
+// SetNameHash - on a MultiResolver
 func (m *MultiResolver) SetNameHash(nameHash func(string) common.Hash) {
 	m.nameHash = nameHash
 }
 
 /*
-Api implements webserver/file system related content storage and retrieval
+API implements webserver/file system related content storage and retrieval
 on top of the FileStore
 it is the public interface of the FileStore which is included in the ethereum stack
 */
-type Api struct {
+type API struct {
 	resource  *mru.Handler
 	fileStore *storage.FileStore
 	dns       Resolver
 }
 
-//the api constructor initialises
-func NewApi(fileStore *storage.FileStore, dns Resolver, resourceHandler *mru.Handler) (self *Api) {
-	self = &Api{
+// NewAPI - the api constructor initialises a new API instance.
+func NewAPI(fileStore *storage.FileStore, dns Resolver, resourceHandler *mru.Handler) (self *API) {
+	self = &API{
 		fileStore: fileStore,
 		dns:       dns,
 		resource:  resourceHandler,
 	}
 	return
 }
 
-// to be used only in TEST
-func (self *Api) Upload(uploadDir, index string, toEncrypt bool) (hash string, err error) {
-	fs := NewFileSystem(self)
+// Upload - to be used only in TEST
+func (a *API) Upload(uploadDir, index string, toEncrypt bool) (hash string, err error) {
+	fs := NewFileSystem(a)
 	hash, err = fs.Upload(uploadDir, index, toEncrypt)
 	return hash, err
 }
 
-// FileStore reader API
-func (self *Api) Retrieve(addr storage.Address) (reader storage.LazySectionReader, isEncrypted bool) {
-	return self.fileStore.Retrieve(addr)
+// Retrieve - FileStore reader API
+func (a *API) Retrieve(addr storage.Address) (reader storage.LazySectionReader, isEncrypted bool) {
+	return a.fileStore.Retrieve(addr)
 }
 
-func (self *Api) Store(data io.Reader, size int64, toEncrypt bool) (addr storage.Address, wait func(), err error) {
+// Store DPA store API
+func (a *API) Store(data io.Reader, size int64, toEncrypt bool) (addr storage.Address, wait func(), err error) {
 	log.Debug("api.store", "size", size)
-	return self.fileStore.Store(data, size, toEncrypt)
+	return a.fileStore.Store(data, size, toEncrypt)
 }
 
+// ErrResolve declaration
```

Comment:
Overall very nice improvement on documenting most of the methods, but comments like this one are useless - I think it would be better to not have them at all.

Reply:
I have to agree with you on this, sometimes the code itself is self evident. But the linter requires some type of comment there. Any suggestions on how we can have both a comment to satisfy the linter and add value in self evident pieces of code like this?

Reply:
Sorry, I didn't know the linter is so strict. I don't have a good comment that will add value... maybe:

```
ErrResolve is returned when an URI cannot be resolved from ENS.
```